source/HailPanel.cpp

```diff
@@ -296,6 +297,12 @@ bool HailPanel::KeyDown(SDL_Keycode key, Uint16 mod, const Command &command, boo
 			player.SetTribute(planet, 0);
 			message = "Thank you for granting us our freedom!";
 		}
+		else if(!planet->IsDefending())
+			GetUI()->Push(new Dialog([this]() { message = planet->DemandTribute(player); },
+				"Demanding tribute may cause this planet to launch defense fleets to fight you. "
+				"After battling the fleets, you can keep demanding tribute until the planet relents.\n"
+				"This barbaric act may hurt your reputation severely. Do you want to proceed?",
+				Truncate::NONE, true, false));
```

Comment:
"Keep demanding tribute" implies to me that you need to repeatedly demand tribute in order for it to succeed, when in reality it's just a single demand to start the battle and another single demand after the battle is over to get the tribute.
```suggestion
		else if(!planet->IsDefending())
			GetUI()->Push(new Dialog([this]() { message = planet->DemandTribute(player); },
				"Demanding tribute may cause this planet to launch defense fleets to fight you. "
				"After battling the fleets, you can demand tribute again for the planet to relent.\n"
				"This act may hurt your reputation severely. Do you want to proceed?",
				Truncate::NONE, true, false));
```

Reply:
I have updated the PR description screenshot with this new dialog.